### ISSUE TITLE
RFC (repost): Priority of algorithm usage across different (Twitter) features

### ISSUE THREAD

**Issue by twister21** [twitter/the-algorithm#9093]
_**tl;dr: fewer algorithms for home feed generation, more algorithms for content & author discovery across the core features Topics, Trends, Lists and Search**_

_Introduction_

X platforms (including Twitter, a **text**-focused service) shouldn't serve a “For You” home feed by default that includes any _embedding space_ (e.g., subscribable creators, tweets or videos “based on your likes”) or a large amount of _social graph_ content, apart from likes and reposts if this hasn't been explicitly disabled, because it isn't suited for users who follow a wide thematic variety of content, and algorithms can't outperform human beings in picking whose _posts_ best fit in one's personal feed. Instead, content moderation is done more effectively by following the right and unfollowing the wrong accounts or topics, but the current feed systems lack customization and discovery features, and need improvement.

Extending curation options and enabling platform interoperability (via shared following lists and histories as well as superfeeds and supersearch) is more reasonable than cloning related platforms, such as YouTube, Substack/Medium, LinkedIn or Discord (https://github.com/orgs/community/discussions/60763). 

_Overview_

Introduce new sorting and filter options for (now multiple) following-only home feeds (see _Feed restructuring_) and provide separate methods to discover follow-worthy accounts and notable _posts_ through better use of a user's

- topic following list, which should be fully viewable via /topics and individually via /t (human-readable identifiers are more useful than generic (such as snowflake) topic ids for direct URL access), also show topic-based|related _followables_ on the sidebar
- view and search history, which should be accessible via /history, and pausable, searchable, selectively deletable (individually, multi-selectable and/or by time period) as well as clearable
- recommendations, which should be filterable by source used for generation and not be language-restricted if auto-translation is enabled

as well as

- general _data record_ similarity (such as Twitter's SimClusters system for accounts), filters (such as topics) and sorting
- account summaries
  - highlight all major topics/subtopics an account posts about (with percentage distribution)
  - allow following (account, topics) pairs
  - auto-generate lists (by account type) based on affiliates
  - link connected accounts of other platforms, and show _post_ type statistics
- search 
  - semantic, multi-language search feature
  - quick access for advanced search
  - additional/separated options (topic: name; account: biography, location; video: description, captions; Space: title) 
  - specifically YouTube: remove unrelated _embedding space_ sections from the results (such as "previously watched", "explore more" or "popular today"), collapse related ones by default, reduce number of "Shorts" sections to 1 (with "see more" button), and improve keyword-based results quality (e.g. the query "youtube" yields lots of results not containing the keyword)
  
_Data feeds_

Feed lists should have _Standard sorting & filter options_, depending on _data record_'s type (_followable_|_post_), and automatically show translations of texts in languages not declared as supported (which can be configured, specifically the primary language, and inferred from an user's own use), unless set to manual mode. For speech audio content, auto-generated language-specific tracks and subtitles should be provided.

_Followables_ (account|topic|community|_list_) feeds should have a multi-follow feature.

Specific feeds (such as replies, tag and individual account) should show _post_ filter group distribution statistics.

_Post_ type can be, for example: 
Twitter: (news) tweet, space
YouTube: (instructive|explanatory|essay|relive|news|interview) video, movie, (podcast|show) episode, short, livestream
Medium: instructive|explanatory|news|interview article
Substack: argumentative|narrative|descriptive essay
LinkedIn: job

Differentiation doesn't make sense if content gets crossposted, which shouldn't be necessary.
Instead, in addition to each platform's type-specific feeds, there should be a filterable super(home|topics|explore|account|...) feed as well as a supersearch feature (via x.com). Similar types across platforms should be treated as identical or be concentrated/moved, e.g., Twitter: tweet; Substack: note; YouTube: post. Perhaps also a super-creation interface.

_List_ (public/private) can be, for example:
_Post_ list: YouTube playlist, Medium list/publication, Twitter bookmark folder
Account list: Twitter list

Differentiate seen count between impressions (served in feed/scrolled-by) and unique account/IP views (except for short-form text content or auto-play, this requires interaction).

Topics are automatically assigned to _data records_, e.g. via Twitter's topic-social-proof service, as well as more specific types to _posts_, if available.

Tags can be manually assigned to _posts_, and need to be verified, at the latest when being served under the tag.

Communities and _lists_ can be edited and assigned human-readable identifiers by permitted accounts.
_Lists_ should be forkable, or _data records_ be submittable.

_Feed restructuring_
**Home**
- "For you" should be replaceable by a "Most interacted" tab, serving (**unseen** and/or _Standard filter options_ respecting) top _posts_ from the account following list.
- Accordingly, "Following" is renamed to "Latest", or introduced if it doesn't exist yet

**Topics**
- Once all content of the home feed has been seen, the (new) custom topics feed should appear
- _Posts_ are listed either grouped by topic or individually (with topic link) based on sorting preference

**Explore**
(Twitter's Explore page currently only shows trends and tweets from five? of the topics a user follows)

- should be enhanced or introduced (see below, _Explore feed_)

This way, users can remove the recommendations from their own timeline and not miss out on highly relevant content without losing the ability to discover novel content. In practice, this can be previewed on a home feed sidebar card titled "Explore more" or "More activity" (instead of "who to follow"), above/below the trends card, if they haven't been hidden.

_Explore feed_
Should consist of automatically curated content:

- trending tags and the **links** of the topics they belong to
- popular (or custom sorted) _data records_ grouped by _Standard filter options_ datetime ranges (expandable preview sections), which should also be filterable by geo-location
- recommended _data records_ based on _social graph_ ("followed or viewed by account(s) you follow") or related/similar to following list as well as like, view or search history

These feeds should be seperately viewable (via /trends, /popular, /recommended), and additionally filterable by _data record_'s type.

_Trends feed_ 
(the term is used differently, in this context: popular tags based on _posts_ in a specific time period (how Twitter uses it),
alternatively: individual popular _posts_ (how e.g. YouTube uses it), ambiguity should be removed by using the same name for the same feature)

- improve grouping ("Trending with" currently only works in some cases)
- quick switch for regional, personal ("For you") and (custom) topic-filtered trends, also selected timespan/historical and international trends (requires translation)
- add option to preview top _post_(s) for each trend
- map events to hashtags, show additional information if available

_(Hash-)Tag feed_
- prevent abuse (incorrectly labeled content) by implementing a semantic filter

_Replies feed_
- replies should additionally be filterable/groupable by
  - spam score/marked as "probable spam"
  - reactions-only (emojis, punctuation marks, interjections, and potentially meme replies)
  - context/recurring tags (replies that refer to the same thing)
  - contains timestamp
- quote _posts_ should be treated as replies  
- reply restrictions for accounts (mentioned, following, verified) should be reader-configurable filters instead, and the options to hide/delete or disable replies become needless
  
This greatly improves the experience, and the original "Community Notes" are replies, curated in a meaningful way.

_Standard sorting & filter options_

sortable by:
- datetime of creation (latest|oldest)
- popularity (top|least relevant): number of
    - _post_: replies + likes + reposts (= engagement), seen count (impressions or views), engagement or interaction/seen count ratio (best)
    - _followable_: followers
      - account: paid followers, affiliates
      - _post_ list: views
      - topic|community (= members): _posts_, authors
- content 
  - length
  - density score

filterable/groupable by (optional):
- datetime range of creation (last 1h|24h|72h|7d|30d|6m|1y|10y; current or selected year; all-time; custom timespan)
- topic(s)
- account: type (personal|organizational|automated); professional category; verified, (inter-)governmental, affiliated, paid-followable (e.g. Twitter: subscribable, YouTube: joinable) (true|false)
- _post_: 
  - (specific) type
  - sentiments (positive|neutra|negative)
  - styles (**formal**: e.g. academic/scientifc, corporate, bureaucratic/legal, not exclusive: journalistic, political|**informal**: e.g. poetic, casual, simple|**profane**)
  - tones (objective|robotic|enthusiastic|passionate|aggressive|annoyed|friendly|confident|diplomatic|humorous|ironic|persuasive)
  - original language
  - content 
       - length (word count; video: duration; text: estimated reading time), tweet: threaded (true|false)
       - types (factual/opinion statement, quote, question, poll, statistic, external link, embedded graphic/video, LaTeX content, code block)
  - only for specific feeds: liked, reposted (true|false)
